integration-tests/opentelemetry-reactive/src/main/java/io/quarkus/it/opentelemetry/reactive/ReactiveResource.java

```diff
@@ -25,10 +27,17 @@ public class ReactiveResource {
     @GET
     public Uni<String> helloGet(@QueryParam("name") String name) {
         Span span = tracer.spanBuilder("helloGet").startSpan();
-        return Uni.createFrom().item("Hello " + name).onItem().delayIt().by(Duration.ofSeconds(2))
+        return Uni.createFrom().item("Hello " + name).onItem().delayIt().by(Duration.ofSeconds(SECONDS_DELAY))
                 .eventually((Runnable) span::end);
     }
 
+    @GET
+    @Path("/hello-get-uni")
+    @WithSpan("helloGetUni")
+    public Uni<String> helloGetUni() {
+        return Uni.createFrom().item("helloGetUni").onItem().delayIt().by(Duration.ofSeconds(SECONDS_DELAY));
```

Comment:
Ah ah, here comes the thread! 
Delay uses a scheduler underneath (the Quarkus scheduler). We do not propagate the duplicated context in these cases (we discussed it, and to avoid leaks we didn't do it).

So, if you want to use delay, either capture the context and use a ContextScheduler (`.delayIt().on(executor).by(...)`) or use the approach I mentioned above using `vertx.timer(...)` - or both to cover every cases.

Reply:
got it

Reply:
done